src/core/plugins/oas3/fn.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * @prettier
+ */
+import { objectify } from "core/utils"
+
+export const makeIsFileUploadIntended = (getSystem) => {
+  const isFileUploadIntended = (schema, mediaType = null) => {
+    const { fileUploadMediaTypes } = getSystem().getConfigs()
+    const isFileUploadMediaType =
+      typeof mediaType === "string" &&
+      fileUploadMediaTypes.some((fileUploadMediaType) =>
+        mediaType.startsWith(fileUploadMediaType)
+      )
+
+    if (isFileUploadMediaType) {
+      return true
+    }
+
+    const { type, format } = objectify(schema)
```

Comment:
Just optimization note: objectify transforms entire schema, which is unnecessary for this case. We're just plucking two top level keywords from the schema, it would more optimal not to convert.